corehq/apps/app_manager/static/app_manager/js/modules/report_module.js

```diff
@@ -396,6 +403,29 @@ hqDefine('app_manager/js/modules/report_module', function () {
             self.reports.push(newReport(currentReports[i]));
         }
 
+        var getInstanceIdsInThisModule = function () {
+            return _.map(self.reports(), function (r) {return r.instanceId();});
+        };
+
+        // flag instance ids with uuids outside this module
+        var uuidsByInstanceId = hqImport('hqwebapp/js/initial_page_data').get('uuids_by_instance_id'),
+            uuidsInThisModule = _.pluck(self.reports(), 'uuid'),
+            instanceIdsElsewhere = _.chain(uuidsByInstanceId)
+                .pairs()
+                .filter(function (idPair) { return _.difference(idPair[1], uuidsInThisModule).length; })
+                .map(_.first)
+                .value();
+
+        self.validateSlug = function (instanceId) {
```

Comment:
`instanceIdsElsewhere` is fixed (although this _could_ query the server to make sure the info is up-to-date), so that should be fine, and the function doesn't use `uuidsInThisModule`.  Are you saying that knockout won't trigger this function at the appropriate times?  Anecdotally, it does seem to work, feel free to play with this test module on staging:
https://staging.commcarehq.org/a/ethan-test/apps/view/ab81ce0c2aaf8df0775ff6712722c72a/module/ae427b9083b972f0bab718e318120c845d1de9b7/
Is there a more "knockout" way to do this?